cloudify_cli/tests/test_import_resolver.py

```diff
@@ -0,0 +1,188 @@
+        import_resolver_config[IMPORT_RESOLVER_KEY][
+            RESLOVER_PARAMETERS_KEY] = parameters
+    return import_resolver_config
+
+
+class GetImportResolverTests(CliCommandTest):
+
+    def setUp(self):
+        super(GetImportResolverTests, self).setUp()
+        self._create_cosmo_wd_settings()
+
+    def test_get_resolver(self):
+        cli_runner.run_cli('cfy init -r')
+        resolver_configuration = create_resolver_configuration(
+            implementation='mock implementation',
+            parameters='mock parameters')
+        update_config_file(resolver_configuration=resolver_configuration)
+        with mock.patch('dsl_parser.utils.create_import_resolver') as \
+                mock_create_import_resolver:
+            utils.get_import_resolver()
+            mock_create_import_resolver.assert_called_once_with(
+                resolver_configuration[IMPORT_RESOLVER_KEY])
+
+    def test_get_custom_resolver(self):
+        cli_runner.run_cli('cfy init -r')
+        parameters = {'param': 'custom-parameter'}
+        custom_resolver_class_path = "%s:%s" % (
+            CustomImportResolver.__module__, CustomImportResolver.__name__)
+        import_resolver_config = create_resolver_configuration(
+            implementation=custom_resolver_class_path, parameters=parameters)
+        update_config_file(resolver_configuration=import_resolver_config)
+        resolver = utils.get_import_resolver()
+        self.assertEqual(type(resolver), CustomImportResolver)
+        self.assertEqual(resolver.param, 'custom-parameter')
+
+
+class ImportResolverLocalUseTests(CliCommandTest):
+
+    def setUp(self):
+        super(ImportResolverLocalUseTests, self).setUp()
+        self._create_cosmo_wd_settings()
+
+    @mock.patch('cloudify_cli.utils.get_import_resolver')
+    def _test_using_import_resolver(self,
+                                    command,
+                                    blueprint_path,
+                                    mock_parse,
+                                    mock_get_resolver):
+        cli_runner.run_cli('cfy init -r')
+
+        parameters = {
+            'rules':
+                [{'rule1prefix': 'rule1replacement'}]
+        }
+        import_resolver_config = create_resolver_configuration(
+            parameters=parameters)
+        update_config_file(import_resolver_config)
+        resolver = DefaultImportResolver(**parameters)
+
+        mock_get_resolver.return_value = resolver
+
+        kwargs = {
+            'dsl_file_path': blueprint_path,
+            'resolver': resolver
+        }
+        cli_command = 'cfy {0} -p {1}'.format(command, blueprint_path)
+        self.assert_method_called(
+            cli_command, mock_parse, 'parse_from_path', kwargs)
+
+    def test_validate_blueprint_uses_import_resolver(self):
+        from cloudify_cli.commands import blueprints
+        blueprint_path = '{0}/local/blueprint.yaml'.format(BLUEPRINTS_DIR)
+        self._test_using_import_resolver(
```

Comment:
how does this work? seems like _test_using_import_resolver expects 4 arguments...